runtime/vm/ValueTypeHelpers.hpp

```diff
@@ -561,6 +561,101 @@ class VM_ValueTypeHelpers {
 		return value;
 	}
 
+	/**
+	 * Copies an array of non-primitive objects
+	 * Handles flattened and non-flattened cases
+	 * A generic special stack frame must be built before calling this function
+	 *
+	 * Assumes srcObject and destObject are not null
+	 * Assumes array bounds (srcIndex to (srcIndex + lengthInSlots), and destIndex to (destIndex + lengthInSlots)) are valid
+	 * Assumes a generic special stack frame has been built on the stack
+	 *
+	 * @param[in] currentThread thread token
+	 * @param[in] objectAccessBarrier access barrier
+	 * @param[in] objectAllocate allocator
+	 * @param[in] srcObject the source array to copy objects from
+	 * @param[out] destObject the destination array in which objects should be stored
+	 * @param[in] srcIndex the index in the source array to begin copying objects from
+	 * @param[in] destIndex the index in the destination array to begin storing objects
+	 * @param[in] lengthInSlots the number of elements to copy
+	 *
+	 * @return 0 if copy was successful, -1 if there was an array store error, -2 if there was a null pointer exception
+	 */
+	static VMINLINE I_32
+	copyFlattenableArray(J9VMThread *currentThread, MM_ObjectAccessBarrierAPI objectAccessBarrier, MM_ObjectAllocationAPI objectAllocate, j9object_t srcObject, j9object_t destObject, I_32 srcIndex, I_32 destIndex, I_32 lengthInSlots)
+	{
+		I_32 srcEndIndex = srcIndex + lengthInSlots;
+		bool typeChecksRequired = false;
+		J9Class *srcClazz = J9OBJECT_CLAZZ(currentThread, srcObject);
+		J9Class *destClazz = J9OBJECT_CLAZZ(currentThread, destObject);
+		J9Class *destComponentClass = ((J9ArrayClass *)destClazz)->componentType;
+
+		/*
+		* Decide if type checks are required
+		* Type checks are not required only if both classes are the same, or the source class is a subclass of the dest class
+		*/
+		if (srcClazz != destClazz) {
+			UDATA srcDepth = J9CLASS_DEPTH(srcClazz);
+			UDATA destDepth = J9CLASS_DEPTH(destClazz);
+			if ((srcDepth <= destDepth) || (destClazz->superclasses[srcDepth] != srcClazz)) {
+				typeChecksRequired = true;
```

Comment:
`typeChecksRequired` is only used at line 639, so this block to set `typeChecksRequired` can be moved into the `else` block at line 628.

Reply:
moved in [25f3eed](https://github.com/eclipse-openj9/openj9/commit/25f3eed6d26349b79515af1233841c0436016b4e)